packages/react/src/FieldSet/FieldSet.test.tsx

```diff
@@ -54,4 +54,36 @@ describe('FieldSet', () => {
 
     expect(ref.current).toBe(component)
   })
+
+  it('renders the provided hint text after the legend', () => {
+    render(<FieldSet legend="Legend" hint="hint text" />)
+
+    const component = screen.getByRole('group', { name: 'Legend (hint text)' })
+
+    expect(component).toBeInTheDocument()
+  })
+
+  it('renders the default hint text after the legend', () => {
+    const { container } = render(<FieldSet legend="Legend" optional={true} />)
```

Comment:
I think we can leave out `={true}`. It doesn't matter much, but maybe users will copy this code. Same for the other tests.

```suggestion
    const { container } = render(<FieldSet legend="Legend" optional />)
```